### PR TITLE
port to nix 0.27

### PR DESCRIPTION
**Pull request by werdahias**

### CODE REVIEW

File: src/input-backend/main.rs [hunk 1]
```diff
@@ -58,7 +59,11 @@ fn main() -> Result<(), zbus::Error> {
 		.udev_assign_seat("seat0")
 		.expect("Could not assign seat0");
 
-	let pollfd = PollFd::new(input.as_raw_fd(), PollFlags::POLLIN);
+
+    let fd = input.as_raw_fd();
+    assert!(fd != -1);
+    let borrowed_fd = unsafe{ BorrowedFd::borrow_raw(input.as_raw_fd())};
+    let pollfd = PollFd::new(&borrowed_fd, PollFlags::POLLIN);
```
Comment (ErikReider) [inline src/input-backend/main.rs:66]: Did the previous impl break anything?

**Review by ErikReider — CHANGES_REQUESTED** [review]
Also formatting borking the Github Action

File: src/input-backend/main.rs (hunk 1 above)
Comment (werdahias) [reply src/input-backend/main.rs:66]: It would not compile otherwise. Note that my rust is not the strongest, this compiled and worked for me.

Comment (werdahias) [reply src/input-backend/main.rs]: @ErikReider now ported to nix 0.29 and with cargo fmt

**Comment by ErikReider** [comment]
Conflicts with main

**Pull request closed without merging**